code/modules/reagents/reagents/medical.dm

```diff
@@ -663,6 +663,47 @@
 /datum/reagent/medicine/quickclot/overdose_crit_process(mob/living/L, metabolism)
 	L.apply_damages(0, 4*REM, 4*REM)
 
+
+/datum/reagent/medicine/quickclotplus
+	name = "Quick Clot Plus"
+	description = "A chemical designed to quickly and painfully remove internal bleeding by encouraging coagulation. Should not be self-administered."
+	color = "#CC00FF"
+	overdose_threshold = REAGENTS_OVERDOSE/5 //Was 4, now 6 //Now 15 //Now 6 again, yay oldQC
+	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL/5 //10u
+	scannable = TRUE
+	custom_metabolism = REAGENTS_METABOLISM * 2.5
+
+/datum/reagent/medicine/quickclotplus/on_mob_life(mob/living/L, metabolism)
+	var/mob/living/carbon/human/H = L
+	for(var/datum/limb/X in H.limbs)
+		for(var/datum/wound/W in X.wounds)
+			if(W.internal)
+				W.damage = max(0, W.damage - (5*REM))
+				X.update_damages()
+				if (X.update_icon())
+					X.owner.UpdateDamageIcon(1)
+	L.reagents.add_reagent(/datum/reagent/toxin,5)
+	L.reagent_shock_modifier -= PAIN_REDUCTION_FULL
+	L.adjustStaminaLoss(30*REM)
+	return ..()
+
+
+/datum/reagent/medicine/quickclotplus/overdose_process(mob/living/L, metabolism)
+	L.apply_damage(6*REM, TOX)
+	if(iscarbon(L))
+		var/mob/living/carbon/C = L
+		if(C.blood_volume < BLOOD_VOLUME_MAXIMUM)
+			C.blood_volume -= 2
+	return ..()
+
+/datum/reagent/medicine/quickclotplus/overdose_crit_process(mob/living/L, metabolism)
+	if(iscarbon(L))
+		var/mob/living/carbon/C = L
+		if(C.blood_volume < BLOOD_VOLUME_MAXIMUM)
+			C.blood_volume -= 20
+	return ..()
```

Comment:
a) Uneeded iscarbon checks
b) why are you calling parent here?

Reply:
Because I copied it verbatim from iron.

Agreed, currently we don't need that check, but there was a time when you COULD inject ayys... and they're carbon too actually, so I have no clue why that's there.

I'll remove the carbon check, and also the one in iron.

Reply:
Aliens are carbon too, and only carbons can process blood sooo

Reply:
Fixed.